src/tcpserver.c

```diff
@@ -190,11 +195,25 @@ static tcplistener_t *tcplistener_create(tcpserver_t *server,
 	listener->cb_conn = cb_conn;
 	listener->cb_recv = cb_recv;
 
-	listener->sd = socket(
-		addr->ai_family,
-		addr->ai_socktype,
-		addr->ai_protocol);
-
+	/**
+	 * not a hot restart, create and bind
+	 */
+	if (rebind) {
+		listener->sd = socket(
+				addr->ai_family,
+				addr->ai_socktype,
+				addr->ai_protocol);
+		snprintf(sd_buffer, 10, "%d", listener->sd);
+		sd_buffer[strlen(sd_buffer)] = '\0';
```

Comment:
This is redundant and would break (since strlen looks for a NULL). Just set `sd_buffer[9] = '\0'`